src/katsdpk8spoc/workflow_controller.py

```diff
@@ -80,55 +83,65 @@ def get_template(self):
+        super().__init__(
+            image=image,
+            name="telstate",
+            template_name="telstate-template",
+            daemon=True
+        )
 
 
 class Ingest(WorkflowStep):
-    def __init__(self, step_id: int = None):
+    def __init__(
+            self, step_id: int = None,
+            image: str="harbor.sdp.kat.ac.za:443/infra/pocingest:0.5"):
         """Ingest step
 
         :param step_id: the unique ingest ID to be assigned to this process
+        :param image: the docker image to be used
         """
-        super().__init__()
-        self.name = f"ingest{step_id}"
-        self.template_name = "ingest-template"
-        self.dependencies = ["telstate"]
-        self.command = ["python"]
-        self.image = "harbor.sdp.kat.ac.za:443/infra/pocingest:0.5"
+        super().__init__(
+            name=f"ingest{step_id}",
+            template_name="ingest-template",
+            dependencies=["telstate"],
+            command=["python"],
+            image=image
+        )
         self.append_argument("./run.sh")
         self.append_argument("-u")
         self.append_named_argument("tasks-telstate-ip", "{{tasks.telstate.ip}}")
 
 
 class Calibrator(WorkflowStep):
-    def __init__(self, step_id: int = None):
+    def __init__(
+            self, step_id: int = None, resources: dict = None,
+            image: str = "harbor.sdp.kat.ac.za:443/infra/poccalibrator:0.1"):
         """Calibrator step
 
         :param step_id: the unique calibrator ID to be assigned to this process
+        :param image: the docker image to be used
         """
-        super().__init__()
-        self.name = f"calibrator{step_id}"
-        self.template_name = "calibrator-template"
+        super().__init__(
+            name=f"calibrator{step_id}",
+            template_name="calibrator-template",
+            dependencies=["telstate"],
+            command=["python"],
+            image=image,
+            resources=resources or {
+                "limits": {
+                    "cpu": "500m",
+                    "memory": "1Gi",
+                    "sdp.kat.ac.za/jellybeans": 1,  # Fake resource
+                },
+                "requests": {"cpu": "500m", "memory": "1Gi",
+                             "sdp.kat.ac.za/jellybeans": 1},
+            }
+        )
         self.append_argument("./run.sh")
         self.append_argument("-u")
         self.append_named_argument("tasks-telstate-ip", "{{tasks.telstate.ip}}")
-        self.dependencies = ["telstate"]
-        self.command = ["python"]
-        self.image = "harbor.sdp.kat.ac.za:443/infra/poccalibrator:0.1"
-        self.resources = {
-            "limits": {
-                "cpu": "500m",
-                "memory": "1Gi",
-                "sdp.kat.ac.za/jellybeans": 1,  # Fake resource
-            },
-            "requests": {"cpu": "500m", "memory": "1Gi", "sdp.kat.ac.za/jellybeans": 1},
-        }
 
 
 class BatchSetup(WorkflowStep):
```

Comment:
Update the BatchSetup as well sa Batch to use the same pattern as the other inits. (i.e. Using the super's init)